File: Java/leetcode75/src/main/java/org/phoenix/leetcode/challenges/Problem31_ReverseLinkedList.java
package org.phoenix.leetcode.challenges;

import org.phoenix.leetcode.challenges.Problem29_DeleteTheMiddleNodeOfALinkedList.ListNode;

public class Problem31_ReverseLinkedList {
    public ListNode reverseList(ListNode head) {
        if (head == null) {
            return null;
        }
        ListNode curr = head, prev = null, next = null;
        while (curr != null) {
            next = curr.next;
            curr.next = prev;
            prev = curr;
            curr = next;
        }
        return prev;
    }
}

File: JavaScript/leetcode75/src/main/Problem31_ReverseLinkedList.js
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function (head) {
  if (head == null) return null;
  let curr = head,
    prev = null;
  while (curr !== null) {
    let next = curr.next;
    curr.next = prev;
    prev = curr;
    curr = next;
  }
  return prev;
};

module.exports = reverseList;
